### PR TITLE
refactored do_vote to write to reddis, via dbActions in persist.js

### PR DESCRIPTION
**Pull request by zactimeinc**
My confidence level in this is not very high, but it works locally.

### CODE REVIEW

**Comment by hgutknecht** [comment]
:+1: for me

File: routes/do_vote.js [hunk 1]
```diff
@@ -33,28 +36,63 @@ exports.post = function (req, res, next) {
...
-      voteMatch = true;
+  dbActions.getActivePollId(getActivePoll);
+  function getActivePoll(pollId) {
+    console.log('Current Active Pollid: ' + pollId);
+    dbActions.getPoll(pollId, setData);
+    activePoll = pollId;
+  }
+
+  function setData(poll_string) {
+    var pollTitleOnly = false;
```
Comment (ben833) [inline routes/do_vote.js:46]: Eventually, the polls will all have proper JSON in them, so we won't need this. Did you do this just to get things going?
This probably stems from the fact that I didn't get to save the poll js object in start_vote, just the pollName.

File: routes/do_vote.js [hunk 2]
```diff
@@ -33,28 +36,63 @@ exports.post = function (req, res, next) {
...
+          voteMatch = true;
+        }
+      });
+      if (!voteMatch) {
+        console.log('No match, creating new poll option for: ' + voteText);
+        newVote = {
+          voteName: voteText,
+          voteCount: 1
+        };
+      data.votes.push(newVote);
```
Comment (ben833) [inline routes/do_vote.js:84]: Indentation

File: routes/do_vote.js [hunk 3]
```diff
@@ -33,28 +36,63 @@ exports.post = function (req, res, next) {
...
+      }
+    }
+    dbActions.setPoll(activePoll, JSON.stringify(data), handleResults);
   }
 
-  // TODO SET ACTIVE POLL WITH NEW RESULTS
+  function handleResults(poll) {
+    slackRes = tally.printPoll(data);
+    voteMatch = false;
```
Comment (ben833) [inline routes/do_vote.js]: Why is this line in there?

File: routes/do_vote.js (hunk 1 above)
Comment (zactimeinc) [inline routes/do_vote.js:46]: Yes, this was just so I could work with start_vote as is and should get removed once that is writing a poll obj.

**Pull request merged**